tasks/main.yml

```diff
@@ -16,26 +16,33 @@
 
 - name: apply kubectl service to create inception service
   command: |
-    {{bin_dir}}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
+    {{kubectl_bin_dir}}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
   with_items: "{{ inception_files }}"
 
-- name: copy over jupyter notebook server config files
+- name: copy over jupyter notebook deployment config
   copy:    
-    src: "{{ item }}"
+    src: k8s-tf-nb-deployment.yml 
     dest: "{{ kube_config_dir }}/"
-  with_items: "{{ tf_nb_files }}"
 
+- name: copy over jupyter notebook service config
+  template:    
+    src: k8s-tf-nb-service.yml.j2
+    dest: "{{ kube_config_dir }}/k8s-tf-nb-service.yml"
 
 - name: apply kubectl service to create notebook service
   command: |
-    {{ bin_dir }}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
+    {{ kubectl_bin_dir }}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
   with_items: "{{ tf_nb_files }}"
 
 - name: wait for inception service to start on port 30000
-  wait_for: port=30000
+  wait_for:
+    host: "{{ hostvars[groups['kube-node'][0]]['ansible_host'] }}"
```

Comment:
We need to somehow query k8s to find the NodePort for the service.

Ideally we'd have a load balancer in front of the service with a single well known IP, but I don't think we're quite there yet.

Reply:
My understanding is that the service is the load balancer. Query the ip of any node and it routes traffic to the correct pods.

Reply:
yeah that makes more sense now. You'd still put an external load balancer in front of the cluster otherwise your magic point of entry node would become a bottleneck and SPOF.